pkg/rest/vf/state_change.go

```diff
@@ -54,40 +54,18 @@ func (vm *VzVirtualMachine) HardStop() error {
 	return vm.VzVM.Stop()
 }
 
-func (vm *VzVirtualMachine) CanChangeState(s define.StateChange) (bool, error) {
-	switch s {
-	case define.Pause:
-		return vm.CanPause(), nil
-	case define.Resume:
-		return vm.CanResume(), nil
-	case define.Stop:
-		return vm.CanStop(), nil
-	case define.HardStop:
-		return vm.CanHardStop(), nil
-	default:
-		return false, fmt.Errorf("invalid state: %s", s)
-	}
-}
-
 func (vm *VzVirtualMachine) CanPause() bool {
-	can := vm.VzVM.CanPause()
-	logrus.Debug("can pause: ", can)
-	return can
+	return vm.VzVM.CanPause()
 }
 
 func (vm *VzVirtualMachine) CanResume() bool {
-	can := vm.VzVM.CanResume()
-	logrus.Debug("can resume: ", can)
-	return can
+	return vm.VzVM.CanResume()
 }
 
 func (vm *VzVirtualMachine) CanStop() bool {
-	can := vm.VzVM.CanRequestStop()
-	logrus.Debug("can stop: ", can)
-	return can
+	return vm.VzVM.CanRequestStop()
 }
 
 func (vm *VzVirtualMachine) CanHardStop() bool {
-	logrus.Debug("can hard top: ", true)
```

Comment:
there's a `vm.vzVM.CanStop()`

Reply:
In fact, our `Stop` corresponds to `RequestStop` in vz, and `HardStop` corresponds to `Stop` in vz. In vz, there is no API called `CanStop` (because powering off is always possible at any time).

Reply:
Good catch. I used to think that vz did not provide this API.😅